src/SVGViewer.tsx

```diff
@@ -607,8 +608,84 @@ function SVGViewer({
     }
   }
 
+  const gridProps = {
+    'x': {
+      min: bounds[0],
+      max: bounds[2],
+      delta: bounds[2] - bounds[0],
+    },
+    'y': {
+      min: bounds[1],
+      max: bounds[3],
+      delta: bounds[3] - bounds[1],
+    }
+  } as any; // TODO TS to fix
+
+  // we need the labels to be always on top of the lines, but since SVG doesn't support z-index via CSS
+  // we need to generate them separately and then push them in the right order in the "data.grid" element
+  const gridElements = {
+    lines: [],
+    labels: [],
+  } as any; // TODO TS to fix
+
+  // TODO: discuss in PR if is really needed, or we can assume they have the same scale
+  // TODO explain what the "power of ten scale" is
+  const gridTenScaleX = Math.pow(10, Math.floor(Math.log10(gridProps['x'].delta)));
+  const gridTenScaleY = Math.pow(10, Math.floor(Math.log10(gridProps['y'].delta)));
+  gridProps['scalePow10'] = Math.min(gridTenScaleX, gridTenScaleY);
+
+  // TODO explain what a "step" is
+  gridProps['step'] = gridProps['scalePow10'] / 5;
+
+  ['x', 'y'].forEach((axis) => {
+    // not very elegant, but it works
+    // TODO alternative, use some "magic" code that uses "object.keys(gridProps) and works out the "other" key ?
+    const otherAxis = axis === 'x' ? 'y' : 'x';
```

Comment:
I'm fine with this, clear code beats magic code, especially when there are only 2 cases haha